File: psra/service/src/main/java/by/arhor/psra/dto/GalleryDto.java
package by.arhor.psra.dto;

import by.arhor.psra.CoreVersion;

import java.util.Objects;
import java.util.StringJoiner;

public final class GalleryDto extends Dto {

  private static final long serialVersionUID = CoreVersion.SERIAL_VERSION_UID;

  private String name;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    GalleryDto that = (GalleryDto) o;
    return Objects.equals(name, that.name);
  }

  @Override
  public int hashCode() {
    return Objects.hash(super.hashCode(), name);
  }

  @Override
  public String toString() {
    return new StringJoiner(", ", GalleryDto.class.getSimpleName() + "[", "]")
        .add("name='" + name + "'")
        .toString();
  }
}

File: psra/service/src/main/scala/by/arhor/psra/dto/GalleryDto.scala
package by.arhor.psra.dto

import java.util.Objects

import scala.beans.BeanProperty

class GalleryDto extends Dto {

  @BeanProperty var name: String = _

  override def equals(obj: Any): Boolean = {
    if (super.equals(obj) && getClass == obj.getClass) {
      val gallery = obj.asInstanceOf[GalleryDto]
      name == gallery.name
    } else {
      false
    }
  }

  override def hashCode(): Int = super.hashCode() + Objects.hash(
    getName
  )

}
